src/EditorFeatures/CSharpTest/ImplementInterface/ImplementExplicitlyTests.cs

```diff
@@ -0,0 +1,478 @@
+    public int M(int i) { }
+    public event Action Ev { add { } remove { } }
+
+    void InternalExplicit()
+    {
+        var v = this.Prop;
+        this.Prop = 1;
+        this.Prop++;
+        ++this.Prop;
+
+        this.M(0);
+        this.M(this.M(0));
+
+        this.Ev += () => {};
+
+        var v1 = nameof(this.Prop);
+    }
+}",
+@"
+using System;
+interface IGoo { int Prop { get; set; } int M(int i); event Action Ev; }
+
+class C : IGoo
+{
+    int IGoo.Prop { get { } set { } }
+    int IGoo.M(int i) { }
+    event Action IGoo.Ev { add { } remove { } }
+
+    void InternalExplicit()
+    {
+        var v = ((IGoo)this).Prop;
+        ((IGoo)this).Prop = 1;
+        ((IGoo)this).Prop++;
+        ++((IGoo)this).Prop;
+
+        ((IGoo)this).M(0);
+        ((IGoo)this).M(((IGoo)this).M(0));
+
+        ((IGoo)this).Ev += () => {};
+
+        var v1 = nameof(((IGoo)this).Prop);
+    }
+}", index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestUpdateReferences_External()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+interface IGoo { int Prop { get; set; } int M(int i); event Action Ev; }
+
+class C : IGoo
+{
+    public int [||]Prop { get { } set { } }
+    public int M(int i) { }
+    public event Action Ev { add { } remove { } }
+}
+
+class T
+{
+    void External(C c)
+    {
+        var v = c.Prop;
+        c.Prop = 1;
+        c.Prop++;
+        ++c.Prop;
+
+        c.M(0);
+        c.M(c.M(0));
+
+        c.Ev += () => {};
+
+        new C
+        {
+            Prop = 1
+        };
+
+        var v1 = nameof(c.Prop);
+    }
+}",
+@"
+using System;
+interface IGoo { int Prop { get; set; } int M(int i); event Action Ev; }
+
+class C : IGoo
+{
+    int IGoo.Prop { get { } set { } }
+    int IGoo.M(int i) { }
+    event Action IGoo.Ev { add { } remove { } }
+}
+
+class T
+{
+    void External(C c)
+    {
+        var v = ((IGoo)c).Prop;
+        ((IGoo)c).Prop = 1;
+        ((IGoo)c).Prop++;
+        ++((IGoo)c).Prop;
+
+        ((IGoo)c).M(0);
+        ((IGoo)c).M(((IGoo)c).M(0));
+
+        ((IGoo)c).Ev += () => {};
+
+        new C
+        {
+            Prop = 1
+        };
+
+        var v1 = nameof(((IGoo)c).Prop);
+    }
+}", index: 1);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsImplementInterface)]
+        public async Task TestMemberWhichImplementsMultipleMembers()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+interface IGoo { int M(int i); }
+interface IBar { int M(int i); }
+
+class C : IGoo, IBar
+{
+    public int [||]M(int i)
+    {
+        throw new System.Exception();
+    }
+}",
+@"
+interface IGoo { int M(int i); }
+interface IBar { int M(int i); }
+
+class C : IGoo, IBar
+{
+    int IGoo.M(int i)
+    {
+        throw new System.Exception();
+    }
+    int IBar.M(int i)
+    {
+        throw new System.Exception();
```

Comment:
TBH, because it's what resharper does.  And if i had to have the others call the first one, i'd have to write all the code to make that work for all the cases :D

Reply:
i think this is acceptable this way for now.